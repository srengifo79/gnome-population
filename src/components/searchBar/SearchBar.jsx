import { useState } from "react";
import styled from "styled-components";

import { colors } from "../../theme/colors";

const StyledSearchBar = styled.div`
  display: flex;
  justify-content: space-between;

  input {
    flex: 1;
    padding: 0.5em;
    color: ${colors.white};
    background-color: ${colors.paleBLue};
    outline: none;
    border: 1px solid ${colors.black};
    border-radius: 0.2em;
    margin-right: 0.5rem;

    &::placeholder {
      color: ${colors.white};
    }
  }

  button {
    color: ${colors.textBlue};
    padding: 0.6em 1em;
    border-radius: 2px;
    background: ${colors.paleBLue};
    border: 1px solid ${colors.black};
  }
`;
const SearchBar = ({ onSearch, className }) => {
  const [searchValue, setSearchValue] = useState("");
  return (
    <StyledSearchBar className={className}>
      <input
        value={searchValue}
        onChange={(e) => setSearchValue(e.target.value)}
        placeholder="Search by name"
      />
      <button onClick={() => onSearch(searchValue)}>Search</button>
    </StyledSearchBar>
  );
};

export default SearchBar;
