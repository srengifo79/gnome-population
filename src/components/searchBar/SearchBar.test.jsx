import React from "react";
import { screen, render, cleanup, fireEvent } from "@testing-library/react";
import { toMatchDiffSnapshot } from "snapshot-diff";

import SearchBar from "./SearchBar";

expect.extend({ toMatchDiffSnapshot });

afterEach(cleanup);

it("Initial snapshop", () => {
  const { asFragment } = render(<SearchBar onSearch={() => {}} />);
  expect(asFragment()).toMatchSnapshot();
});

it("Input and search click", () => {
  const mockSearch = jest.fn();

  const { asFragment } = render(<SearchBar onSearch={mockSearch} />);

  const firstRender = asFragment();

  const textInput = screen.getByPlaceholderText("Search by name");
  const searchButton = screen.getByText("Search");

  fireEvent.change(textInput, { target: { value: "Testing..." } });
  fireEvent.click(searchButton);

  expect(mockSearch).toBeCalledWith("Testing...");
  expect(firstRender).toMatchDiffSnapshot(asFragment());
});
