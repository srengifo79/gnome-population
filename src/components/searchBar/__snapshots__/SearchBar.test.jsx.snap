// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Initial snapshop 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL krySSK"
  >
    <input
      placeholder="Search by name"
      value=""
    />
    <button>
      Search
    </button>
  </div>
</DocumentFragment>
`;

exports[`Input and search click 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -2,11 +2,11 @@
    <div
      class=\\"sc-bdvvtL krySSK\\"
    >
      <input
        placeholder=\\"Search by name\\"
-       value=\\"\\"
+       value=\\"Testing...\\"
      />
      <button>
        Search
      </button>
    </div>"
`;
